akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -234,10 +234,17 @@ object Behavior {
    */
   def interpretSignal[T](behavior: Behavior[T], ctx: TypedActorContext[T], signal: Signal): Behavior[T] = {
     val result = interpret(behavior, ctx, signal, isSignal = true)
+
+    def isIntercepted: Boolean = result match {
+      case intercepted: InterceptorImpl[T, Any] @unchecked =>
+        intercepted.nestedBehavior._tag == BehaviorTags.ExtensibleBehavior
+      case _ => false
+    }
+
     // we need to throw here to allow supervision of deathpact exception
     signal match {
-      case Terminated(ref) if result == BehaviorImpl.UnhandledBehavior => throw DeathPactException(ref)
-      case _                                                           => result
+      case Terminated(ref) if result == BehaviorImpl.UnhandledBehavior || isIntercepted => throw DeathPactException(ref)
```

Comment:
Hmm, Gut feeling is that it should other have the exact same behavior here and make sure the event sourced behavior returns unhandled for Terminated.

Reply:
I think this is down the wrong path. Looks like the problem is that we handle all signals so the partial function is always matching. For example here: https://github.com/akka/akka/blob/2a812b7dc6ca014f6be60e31d14e55e25f34080c/akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala#L113

I think we need something like:
```
  def isSignalDefined(state: S, signal: Signal): Boolean =
    signalHandler.isDefinedAt((state, signal))
```

so that the partial functions can be written as
```
case signal if setup.isSignalDefined(state.state, signal) =>
        setup.onSignal(state.state, signal, catchAndLog = false)
```

Reply:
The main thing I was wondering is how the nested behavior - Running.Handling.. - in that case can be read as unhandled to fit into that. Which I've not yet found.

Reply:
The other concern which led to putting the handling where it is versus https://github.com/akka/akka/blob/2a812b7dc6ca014f6be60e31d14e55e25f34080c/akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala#L113 is when the `ChildFailed` signal hits that case it starts the much longer train of failure and recovery handling that non-persistent typed does not have, and not wanting to replace one bug with another in that process by handling too soon and missing a step. I'll test @patriknw's suggestion for both cases, thanks.